aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1772,6 +1826,26 @@ impl VMAdapter for AptosVM {
             },
         })
     }
+
+    fn execute_single_transaction_sequential(
```

Comment:
It could live inside vm_wrapper as well. When you explained the logic to me initially, you asked me to implement it this way. I guess we are now going to have re-execution logic in exector.rs instead?

Reply:
I thought initially it was in the wrapper, trait has different implications (for anything implementing that trait) that's why I was asking, but indeed this comment isn't relevant as we move the logic all the way to the executor (which makes sense for having materialization logic there and for testing)